internal/handlers/common/filter.go

```diff
@@ -1025,14 +1034,14 @@ func filterFieldValueByTypeCode(fieldValue any, code typeCode) (bool, error) {
 	}
 
 	switch code {
-	case typeCodeObject:
-		if _, ok := fieldValue.(*types.Document); !ok {
-			return false, nil
-		}
 	case typeCodeArray:
 		if _, ok := fieldValue.(*types.Array); !ok {
 			return false, nil
 		}
+	case typeCodeObject:
```

Comment:
Object/document should go before array. I guess that change is unrelated and not caught by the linter?